<?xml version="1.0"?>
<launch>
    <!-- logging param -->
    <arg name="robot_name" default="robot_1"/>
    <arg name="output_dir" default="$(env HOME)/bratasena_ws/src/soccer_robot_logs"/>
    <arg name="sampling_rate" default="20"/>
    
    <!-- display info -->
    <param name="logging_info" command="echo 'Starting Soccer Robot Data Logger for $(arg robot_name)'" type="str"/>
    
    <!-- main logger node -->
    <node name="soccer_robot_logger" 
          pkg="data_logger" 
          type="soccer_robot_logger.py" 
          output="screen" 
          respawn="false">
        
        <!-- node param -->
        <param name="robot_name" value="$(arg robot_name)"/>
        <param name="output_dir" value="$(arg output_dir)"/>
        <param name="sampling_rate" value="$(arg sampling_rate)"/>
        
    </node>
    
</launch>